chromium_src/chrome/browser/ui/browser_tabrestore.cc

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "chrome/browser/ui/browser_tabrestore.h"
+
+#include "base/strings/utf_string_conversions.h"
+#include "brave/browser/ui/tabs/brave_tab_strip_model.h"
+#include "brave/browser/ui/tabs/features.h"
+#include "brave/browser/ui/tabs/public/constants.h"
+#include "chrome/browser/ui/tab_ui_helper.h"
+#include "chrome/browser/ui/tabs/public/tab_features.h"
+
+#define AddRestoredTab AddRestoredTab_ChromiumImpl
+#define ReplaceRestoredTab ReplaceRestoredTab_ChromiumImpl
+
+#include <chrome/browser/ui/browser_tabrestore.cc>
+
+#undef ReplaceRestoredTab
+#undef AddRestoredTab
+
+namespace chrome {
+
+namespace {
+
+void MaybeRestoreCustomTitleForTab(
+    BraveTabStripModel* model,
+    int tab_index,
+    const std::map<std::string, std::string>& extra_data) {
+  if (!base::FeatureList::IsEnabled(tabs::features::kBraveRenamingTabs)) {
+    return;
+  }
+
+  if (!extra_data.contains(tabs::kBraveTabCustomTitleExtraDataKey) ||
+      extra_data.at(tabs::kBraveTabCustomTitleExtraDataKey).empty()) {
+    return;
+  }
+
+  model->SetCustomTitleForTab(
+      tab_index,
+      base::UTF8ToUTF16(extra_data.at(tabs::kBraveTabCustomTitleExtraDataKey)));
+}
+
+}  // namespace
+
+content::WebContents* AddRestoredTab(
```

Comment:
We potentially need tests some test coverage to these overrides.

Reply:
Sorry, I was a bit confused about which test would that be.

Reply:
`BraveTabStripModelRenamingTabBrowserTest.                       RestoringTabWithCustomTitle` should cover it.